python/ql/src/semmle/python/frameworks/Flask.qll

```diff
@@ -15,127 +15,84 @@ private import semmle.python.ApiGraphs
-  // flask
-  // ---------------------------------------------------------------------------
-  /** Gets a reference to the `flask` module. */
-  API::Node flask() { result = API::moduleImport("flask") }
-
-  /**
-   * Gets a reference to the attribute `attr_name` of the `flask` module.
-   */
-  private API::Node flask_attr(string attr_name) { result = flask().getMember(attr_name) }
-
-  /** Provides models for the `flask` module. */
-  module flask {
-    /** Gets a reference to the `flask.request` object. */
-    API::Node request() { result = flask_attr("request") }
-
-    /** Gets a reference to the `flask.make_response` function. */
-    API::Node make_response() { result = flask_attr("make_response") }
-
+module Flask {
+  /** Provides models for flask view classes (defined in the `flask.views` module) */
+  module Views {
     /**
-     * Provides models for the `flask.Flask` class
+     * Provides models for the `flask.views.View` class and subclasses.
      *
-     * See https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.
+     * See https://flask.palletsprojects.com/en/1.1.x/views/#basic-principle.
      */
-    module Flask {
-      /** Gets a reference to the `flask.Flask` class. */
-      API::Node classRef() { result = flask().getMember("Flask") }
-
-      /** Gets a reference to an instance of `flask.Flask` (a flask application). */
-      API::Node instance() { result = classRef().getReturn() }
-
-      /**
-       * Gets a reference to the attribute `attr_name` of an instance of `flask.Flask` (a flask application).
-       */
-      private API::Node instance_attr(string attr_name) { result = instance().getMember(attr_name) }
-
-      /** Gets a reference to the `route` method on an instance of `flask.Flask`. */
-      API::Node route() { result = instance_attr("route") }
-
-      /** Gets a reference to the `add_url_rule` method on an instance of `flask.Flask`. */
-      API::Node add_url_rule() { result = instance_attr("add_url_rule") }
-
-      /** Gets a reference to the `make_response` method on an instance of `flask.Flask`. */
-      // HACK: We can't call this predicate `make_response` since shadowing is
-      // completely disallowed in QL. I added an underscore to move things forward for
-      // now :(
-      API::Node make_response_() { result = instance_attr("make_response") }
-
-      /**
-       * Gets a reference to the `response_class` attribute on the `flask.Flask` class or an instance.
-       *
-       * See https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.response_class
-       */
-      API::Node response_class() { result = [classRef(), instance()].getMember("response_class") }
+    module View {
+      /** Gets a reference to the `flask.views.View` class or any subclass. */
+      API::Node subclassRef() {
+        result =
+          API::moduleImport("flask")
+              .getMember("views")
+              .getMember(["View", "MethodView"])
+              .getASubclass*()
```

Comment:
Why is `"MethodView"` included here? Should the QLDoc be updated?

Reply:
fixed :+1: